File: Beginner/decInc/decInc.c
#include <stdio.h>

int main(void){
	int N;

	while(1){
		scanf("%d",&N);

		if(N<0 || N>1000){
			printf("N must be 0 <= N <= 1000\n");
		}else{
			break;
		}
	}

	if(N%4==0){
		printf("%d\n",N+1);
	}else{
		printf("%d\n",N-1);
	}
	
	return 0;
}

File: Beginner/decInc/decInc.java
import java.util.Scanner;

public class decInc{
	public static void main(String[] args) {
		int N;
		Scanner scan = new Scanner(System.in);

		while(true){
			N = scan.nextInt();

			if(N<0 || N>1000){
				System.out.println("N must be 0 <= N <= 1000");
			}else{
				break;
			}
		}

		if(N%4==0){
			System.out.println(N+1);
		}else{
			System.out.println(N-1);
		}
	}
}